### PR TITLE
CI: Run 2D and 1D Tests

### PR DESCRIPTION
**Pull request by ax3l**
## Summary

Run tests also for 2D and 1D builds of AMReX.

## Additional background

Related to 2D issues with `DenseBin` in WarpX: https://github.com/ECP-WarpX/WarpX/pull/2550

## Checklist

The proposed changes:
- [x] fix a bug or incorrect behavior in AMReX
- [x] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] include documentation in the code and/or rst files, if appropriate

### CODE REVIEW

File: .github/workflows/gcc.yml [hunk 1]
```diff
@@ -40,8 +40,8 @@ jobs:
         ctest --output-on-failure
 
   # Build libamrex and all tests
-  tests_build:
-    name: GNU@7.5 C++14 Debug Fortran [tests]
+  tests_build_3D:
+    name: GNU@7.5 C++14 3D Debug Fortran [tests]
     runs-on: ubuntu-18.04
     env: {CXXFLAGS: "-fno-operator-names -Werror -Wall -Wextra -Wpedantic -Wnull-dereference -Wfloat-conversion -Wshadow -Woverloaded-virtual -O1"}
```
Comment (PhilMiller) [inline .github/workflows/gcc.yml:46]: I don't see a `-g` here.

Also, the recommendation from GCC for "optimize without impeding debugging" is `-Og`

Comment (PhilMiller) [reply .github/workflows/gcc.yml:46]: Oh, I see, it's a CMake build, in Debug mode.

Comment (ax3l) [reply .github/workflows/gcc.yml:46]: Yep, that's added in `-DCMAKE_BUILD_TYPE=Debug`

File: .github/workflows/gcc.yml [hunk 2]
```diff
@@ -51,18 +51,63 @@ jobs:
...
+            -DAMReX_SPACEDIM=3
+        cmake --build build -j 2
 
-        ctest --output-on-failure
+        ctest --test-dir build --output-on-failure
+
+  tests_build_2D:
+    name: GNU@7.5 C++14 2D Debug Fortran [tests]
+    runs-on: ubuntu-18.04
+    env: {CXXFLAGS: "-fno-operator-names -Werror -Wall -Wextra -Wpedantic -Wnull-dereference -Wfloat-conversion -Wshadow -Woverloaded-virtual -O1"}
```
Comment (ax3l) [inline .github/workflows/gcc.yml]: Skip temporarily:
```suggestion
    env: {CXXFLAGS: "-fno-operator-names -Wall -Wextra -Wpedantic -Wnull-dereference -Wfloat-conversion -Wshadow -Woverloaded-virtual -O1"}
      # -Werror temporarily skipped until we have functional testing established
```

File: .github/workflows/gcc.yml [hunk 3]
```diff
@@ -51,18 +51,63 @@ jobs:
...
+            -DAMReX_PARTICLES=ON        \
+            -DAMReX_SPACEDIM=2
+        cmake --build build -j 2
+
+        ctest --test-dir build --output-on-failure
+
+  tests_build_1D:
+    name: GNU@7.5 C++14 1D Debug Fortran [tests]
+    runs-on: ubuntu-18.04
+    env: {CXXFLAGS: "-fno-operator-names -Werror -Wall -Wextra -Wpedantic -Wnull-dereference -Wfloat-conversion -Wshadow -Woverloaded-virtual -O1"}
```
Comment (ax3l) [inline .github/workflows/gcc.yml]:
```suggestion
    env: {CXXFLAGS: "-fno-operator-names -Wall -Wextra -Wpedantic -Wnull-dereference -Wfloat-conversion -Wshadow -Woverloaded-virtual -O1"}
      # -Werror temporarily skipped until we have functional testing established
```

**Comment by atmyers** [comment]
Added a missing `AMREX_D_DECL` - let's see if the CI passes now.

**Comment by ax3l** [comment]
Thanks a lot! Found one more, let's see now :crossed_fingers: :slightly_smiling_face:

**Comment by ax3l** [comment]
1D is still a bit tricky it seems.

**Comment by ax3l** [comment]
Getting there, what's still failing in 2D:
```
The following tests FAILED:
	  1 - AsyncOut_multifab (Failed) - fixed now
	 14 - Particles_NeighborParticles (Failed) - fixed now
	 15 - Particles_NeighborParticles_MPI (Failed) - fixed now
	 16 - Particles_ParallelContext (Failed) - fixed now
	 17 - Particles_ParallelContext_MPI (Failed) - fixed now
	 20 - Particles_ParticleMesh (Failed) - fixed now
	 21 - Particles_ParticleMeshMultiLevel (Failed) - fixed now
	 22 - Particles_ParticleReduce (Failed) - fixed now
	 23 - Particles_ParticleTransformations (Failed) - fixed now
```

**Comment by ax3l** [comment]
Wuhu, 2D passes now :tada: 

1D leftovers:
```
The following tests FAILED:
	  2 - MultiBlock_Advection (Failed)
	  3 - MultiBlock_IndexType (Failed)
	 25 - LinearSolvers_NodalPoisson (Failed)
```

**Comment by WeiqunZhang** [comment]
We should simply skip those tests, because they are not meant to work in 1D.

**Review by atmyers — APPROVED** [review]

**Comment by WeiqunZhang** [comment]
Because of a name change in this PR, the required test `GNU@7.5 C++14 Debug Fortran [tests]` no longer exists.  I have been trying to delete that from the branch protection rule since yesterday.  But unfortunately, Github always says
```
Ooops!!! 500

Looks like something went wrong!

We track these errors automatically, but if the problem persists feel free to contact us. In the meantime, try refreshing.
```

@atmyers Could you try?  Maybe it's just me.

**Comment by atmyers** [comment]
Ha, I have been trying this too, including just a moment ago. Not sure what's going wrong.

**Comment by atmyers** [comment]
Looks like it's now letting me edit the branch protection rules again.

**Pull request merged**